src/prefect/engine/cloud/task_runner.py

```diff
@@ -148,3 +149,42 @@ def initialize_run(  # type: ignore
         context.update(cloud=True)
 
         return super().initialize_run(state=state, context=context)
+
+    @call_state_handlers
+    def check_task_is_cached(self, state: State, inputs: Dict[str, Result]) -> State:
+        """
+        Checks if task is cached in the DB and whether any of the caches are still valid.
+
+        Args:
+            - state (State): the current state of this task
+            - inputs (Dict[str, Result]): a dictionary of inputs whose keys correspond
+                to the task's `run()` arguments.
+
+        Returns:
+            - State: the state of the task after running the check
+
+        Raises:
+            - ENDRUN: if the task is not ready to run
+        """
+        if self.task.cache_for is not None:
+            oldest_valid_cache = datetime.datetime.utcnow() - self.task.cache_for
+            cached_states = self.client.get_latest_cached_states(
+                task_id=self.task.id, created_after=oldest_valid_cache
+            )
+
+            for candidate_state in cached_states:
+                assert isinstance(candidate_state, Cached)  # mypy assert
+                if self.task.cache_validator(
+                    candidate_state, inputs, prefect.context.get("parameters")
+                ):
+                    candidate_state._result = candidate_state._result.to_result()
+                    return candidate_state
+
+            self.logger.debug(
```

Comment:
Not functionality related at all, but this message will be logged even if there are *no* cached_states available, which could be confusing. For example, on the very first run of this task it would print that "none of the caches were valid".

Reply:
True.  I'll update!